test_spack.py

```diff
@@ -0,0 +1,214 @@
+            && spack devbuildcosmo cosmo@dev-build%pgi cosmo_target=gpu +cppdycore \
+            && cd .. && rm -rf cosmo', # Listed in https://c2sm.github.io/spack-c2sm/QuickStart.html 
+        'git clone git@github.com:MeteoSwiss-APN/cosmo.git \
+            && cd cosmo \
+            && spack devbuildcosmo cosmo@dev-build%pgi cosmo_target=cpu ~cppdycore \
+            && cd .. && rm -rf cosmo', # Listed in https://c2sm.github.io/spack-c2sm/QuickStart.html
+        # 'spack installcosmo cosmo@master%pgi cosmo_target=gpu +cppdycore ^eccodes +aec +build_shared_libs',
+    },
+    'cosmo-dycore' : {},
+    'cosmo-eccodes-definitions' : {},
+    'cosmo-grib-api' : {},
+    'cosmo-grib-api-definitions' : {},
+    'cuda' : {},
+    'dawn' : {},
+    'dawn4py' : {},
+    'dusk' : {},
+    'dyicon_benchmarks' : {},
+    'ecbuild' : {},
+    'eccodes' : {},
+    'eckit' : {},
+    'gridtools' : {},
+    'icon' : {
+        # 'git clone --recursive git@gitlab.dkrz.de:icon/icon-cscs.git \
+        #     && cd icon-cscs \
+        #     && mkdir pgi_cpu \
+        #     && cd pgi_cpu \
+        #     && touch a_fake_file.f90 \
+        #     && spack dev-build -u build icon@dev-build%pgi config_dir=./.. icon_target=cpu', # Listed in https://c2sm.github.io/spack-c2sm/QuickStart.html
+        # 'git clone --recursive git@gitlab.dkrz.de:icon/icon-cscs.git \
+        #     && cd icon-cscs \
+        #     && mkdir pgi_gpu \
+        #     && cd pgi_gpu \
+        #     && touch a_fake_file.f90 \
+        #     && spack dev-build -u build icon@dev-build%pgi config_dir=./.. icon_target=gpu', # Listed in https://c2sm.github.io/spack-c2sm/QuickStart.html
+    },
+    'icondusk-e2e' : {},
+    'icontools' : {},
+    'int2lm' : {
+        'spack install int2lm@c2sm_master%pgi', # Listed in https://c2sm.github.io/spack-c2sm/QuickStart.html
+        'spack install int2lm@org_master%pgi pollen=False', # Listed in https://c2sm.github.io/spack-c2sm/QuickStart.html
+    },
+    'libgrib1' : {},
+    'mpich' : {},
+    'oasis' : {},
+    'omnicompiler' : {},
+    'omni-xmod-pool' : {},
+    'openmpi' : {},
+    'serialbox' : {},
+    'xcodeml-tools' : {},
+    'zlib_ng' : {},
+}
+
+def UseCases(packages : set):
+    return {case for p in packages for case in use_cases[p]}
+
+# Maps packages to Set[packages it depends on].
+dependencies = {
+    'atlas' : {'ecbuild', 'eckit'},
+    'atlas_utilities' : {'atlas', 'eckit'},
+    'claw' : {},
+    'cosmo' : {'cuda', 'serialbox', 'libgrib1', 'cosmo-grib-api-definitions', 'cosmo-eccodes-definitions', 'omni-xmod-pool', 'claw', 'zlib_ng'},
+    'cosmo-dycore' : {'gridtools', 'serialbox', 'cuda'},
+    'cosmo-eccodes-definitions' : {'eccodes'},
+    'cosmo-grib-api' : {},
+    'cosmo-grib-api-definitions' : {'cosmo-grib-api'},
+    'cuda' : {},
+    'dawn' : {},
+    'dawn4py' : {},
+    'dusk' : {'dawn4py'},
+    'dyicon_benchmarks' : {'atlas_utilities', 'atlas', 'cuda'},
+    'ecbuild' : {},
+    'eccodes' : {},
```

Comment:
eccodes depends on eckit and ecbuild

Reply:
uhmm my bad, it does not depend on eckit indeed. 
Still it depends on ecbuild, 
https://github.com/ecmwf/eccodes/blob/82a4d1ff748cabe722375d311a82e49e60442103/CMakeLists.txt#L21
but it seems that when they create the releases
https://software.ecmwf.int/wiki/download/attachments/45757960/eccodes-2.2.0-Source.tar.gz?api=v2

they ship ecbuild inside the eccodes package. So at the end there is no dependency for spack to handle